packages/playwright-core/src/server/happy-eyeballs.ts

```diff
@@ -126,3 +127,11 @@ async function lookupAddresses(hostname: string): Promise<dns.LookupAddress[]> {
   return result;
 }
 
+function clientRequestArgsToHostName(options: http.ClientRequestArgs): string {
+  if (options.hostname)
+    return options.hostname;
+  if (options.host)
```

Comment:
There is some dubious logic around hostname / host etc inside http client option processing. Let's double check that it doesn't break with urls that have ip as the host.

Reply:
Looks similar to how we have it:

https://github.com/nodejs/node/blob/f69e84c75f8fbf48e5019af6d8b59e949c5cd653/lib/_http_client.js#L191-L193